### PR TITLE
hubble: bump protoc{,-gen-go} and dependencies

### PR DESCRIPTION
**Pull request by rolinh**
This PR bumps `protoc` from 3.12.4 to 3.17.3 and `protoc-gen-go` from 1.23.0 to 1.27.1. In turn, this means that gRPC-Go v1.32.0 or later is now required. PR #15123 updated gRPC to v1.39.0 which means this is not a problem for Cilium anymore. In turn, this means that the new gRPC dial option `WithReturnConnectionError()` can be used which means that Hubble Relay will finally get descriptive errors when connection attempts to peers fail. See commits for details.

### CODE REVIEW

**Comment by rolinh** [comment]
test-me-please

File: api/v1/Makefile.protoc [hunk 1]
```diff
@@ -30,7 +30,7 @@ all:
 		echo Generating $${proto} && \
 		$(PROTOC) $(HUBBLE_PROTOC_PLUGINS) -I $(HUBBLE_PROTO_PATH) \
 			--go_out=paths=source_relative:. \
-			--go-grpc_out=requireUnimplementedServers=false,paths=source_relative:. \
+			--go-grpc_out=require_unimplemented_servers=false,paths=source_relative:. \
```
Comment (glibsm) [inline api/v1/Makefile.protoc:33]: 😕

**Review by glibsm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
Nice :tada:

**Comment by rolinh** [comment]
test-runtime

**Comment by tklauser** [comment]
Depending on whether this PR or #16866 gets merged first, the respective other PR needs to update its `cilium-builder` image again.

**Review by nathanjsweet — APPROVED** [review]
Boom! I call it good.

**Review by errordeveloper — APPROVED** [review]

**Comment by rolinh** [comment]
test-me-please

**Pull request merged**